src/game.cpp

```diff
@@ -11724,6 +11724,49 @@ void game::place_player( const tripoint &dest_loc )
     // Important: don't use dest_loc after this line. `update_map` may have shifted the map
     // and dest_loc was not adjusted and therefor is still in the un-shifted system and probably wrong.
 
+    //Auto pulp or butcher
+    const std::string pulp_butcher = get_option<std::string>( "AUTO_PULP_BUTCHER" );
+    if( pulp_butcher != "off" && ( !get_option<bool>( "AUTO_PULP_BUTCHER_SAFEMODE" ) || mostseen == 0 ) ) {
+        if ( pulp_butcher == "butcher" && u.max_quality( quality_id( "BUTCHER" ) ) > INT_MIN ) {
+            std::vector<int> corpses;
+            auto items = m.i_at(u.pos());
+
+            for( size_t i = 0; i < items.size(); i++ ) {
+                if ( items[i].is_corpse() ) {
+                    corpses.push_back(i);
+                }
+            }
+
+            if( !corpses.empty() ) {
+                u.assign_activity( activity_id( "ACT_BUTCHER" ), 0, -1 );
+                for( int i : corpses ) {
+                    u.activity.values.push_back( i );
+                }
+            }
+        } else if ( pulp_butcher == "pulp" || pulp_butcher == "pulp_adjacent" ) {
+            static auto pulp = [&]( const tripoint &pos ) {
```

Comment:
As far as I recall, lambdas should be declared constant where possible, otherwise they may be treated differently by the compiler or fail to compile on older compilers.